app/models/friend_request.rb

```diff
@@ -1,4 +1,27 @@
 class FriendRequest < ApplicationRecord
-  belongs_to :requestor, class_name: :User
-  belongs_to :receiver, class_name: :User
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  validates :user, presence: true
+  validates :friend, presence: true, uniqueness: { scope: :user }
+  validate :not_self
+  validate :not_friends
+  validate :not_pending
+
+  def accept
+    user.friends << friend
+    destroy
```

Comment:
I think, we shouldn't destroy friendrequest object,
Better solution would be to have status in FriendRequest and make it `pending`, `accepted` , `rejected`,
So, that we will have history as well.

Reply:
Will work on this issue in latest branch